dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -358,3 +251,119 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
         }
     }
 }
+
+namespace winrt::Microsoft::Windows::PushNotifications::factory_implementation
+{
+
+    inline constexpr auto c_maxBackoff{ 5min };
+    inline constexpr auto c_initialBackoff{ 60s };
+    inline constexpr auto c_backoffIncrement{ 60s };
+
+    const HRESULT WNP_E_NOT_CONNECTED = static_cast<HRESULT>(0x880403E8L);
+    const HRESULT WNP_E_RECONNECTING = static_cast<HRESULT>(0x880403E9L);
+    const HRESULT WNP_E_BIND_USER_BUSY = static_cast<HRESULT>(0x880403FEL);
+
+    bool PushNotificationManager::IsChannelRequestRetryable(const hresult& hr)
+    {
+        switch (hr)
+        {
+        case HRESULT_FROM_WIN32(ERROR_TIMEOUT):
+        case WNP_E_NOT_CONNECTED:
+        case WPN_E_OUTSTANDING_CHANNEL_REQUEST:
+        case WNP_E_RECONNECTING:
+        case WNP_E_BIND_USER_BUSY:
+        case HRESULT_FROM_WIN32(RPC_S_SERVER_UNAVAILABLE):
+            return true;
+        default:
+            return false;
+        }
+    }
+
+    winrt::Windows::Foundation::IAsyncOperationWithProgress<winrt::Microsoft::Windows::PushNotifications::PushNotificationCreateChannelResult, winrt::Microsoft::Windows::PushNotifications::PushNotificationCreateChannelStatus> PushNotificationManager::CreateChannelAsync(winrt::guid const& remoteId)
+    {
+        try
+        {
+            auto strong_this{ get_strong() };
```

Comment:
Is this line the only change from the original implementation?

Reply:
Yup. The only change to the implementation that was needed was to get a strong reference.

Reply:
Gotcha. Makes sense.

Reply:
Alternatively, you can also use the get_weak pattern in the caller API